lean4-infoview/src/infoview/messages.tsx

```diff
@@ -0,0 +1,119 @@
+import * as React from 'react';
+import fastIsEqual from 'react-fast-compare';
+import { Location, DocumentUri, DiagnosticSeverity } from 'vscode-languageserver-protocol';
+
+import { basename, escapeHtml, colorizeMessage, RangeHelpers, usePausableState, useEvent, addUniqueKeys, DocumentPosition } from './util';
+import { LeanDiagnostic } from '../lspTypes';
+import { ConfigContext, EditorContext, DiagnosticsContext } from './contexts';
+import { Details } from './collapsing';
+
+interface MessageViewProps {
+    uri: DocumentUri;
+    diag: LeanDiagnostic;
+}
+
+const MessageView = React.memo(({uri, diag}: MessageViewProps) => {
+    const ec = React.useContext(EditorContext);
+    const fname = escapeHtml(basename(uri));
+    const {line, character} = diag.range.start;
+    const loc: Location = { uri, range: diag.range };
+    const shouldColorize = diag.severity === DiagnosticSeverity.Error;
+    let text = escapeHtml(diag.message)
+    text = shouldColorize ? colorizeMessage(text) : text;
+    const severityClass: {[name: number]: string} = diag.severity ? {
```

Comment:
```suggestion
    const severityClass: {[name: number]: string} = diag.severity ? {
```
This type annotation is almost certainly not what is intended.  (Surprisingly it is still correct though, because every string is also a {[_:number]: string}.)